templates/jobs/check_changelog.yml

```diff
@@ -5,7 +5,19 @@
       if [ -z "$CI_MERGE_REQUEST_DIFF_BASE_SHA" ]; then
         exit 0
       fi
-    - git --no-pager diff --name-only "$(git merge-base "$CI_COMMIT_SHA" "$CI_MERGE_REQUEST_DIFF_BASE_SHA")" "$CI_COMMIT_SHA" | grep CHANGELOG.md || exit_code=$?
+    # If the branch is a fix the check will be skipped
+    - |
+      if [ -n "$(echo "${CI_MERGE_REQUEST_SOURCE_BRANCH_NAME}" | grep -E "^fix\/.+")" ]; then
+        exit 0
+      fi
+    # If the pipeline is running for a tag, we check if the tag is present in the changelog
+    # otherwise the CHANGELOG.md should be edited
+    - |
+      if [ -n "${CI_COMMIT_TAG}" ]; then
+        grep ${CI_COMMIT_TAG} CHANGELOG.md || exit_code=$?
+      else
+        git --no-pager diff --name-only "$(git merge-base "$CI_COMMIT_SHA" "$CI_MERGE_REQUEST_DIFF_BASE_SHA")" "$CI_COMMIT_SHA" | grep CHANGELOG.md || exit_code=$?
```

Comment:
where is `$exit_code` used?

Reply:
it is used [here[(https://github.com/sparkfabrik/spark-k8s-deployer/pull/58/files#diff-224917d8fa47ad42f863a94f931217e867006de389bc4550636bb22229841698R22) to print the pipeline fails

Reply:
Ah, down below, I'm a moron 😄